### PR TITLE
refactor: use CGW SDK enums

### PR DESCRIPTION
**Pull request by DiogoSoaress**
## What it solves
Updates to CGW SDK latest version using the added types:

- ConflictType
- DetailedExecutionInfoType
- TransactionInfoType
- TransactionListItemType

## How to test it
History and Queue transaction lists should work as before
Expanding any transaction type details should work as before

### CODE REVIEW

File: src/utils/transaction-guards.ts [hunk 1]
```diff
@@ -47,32 +54,15 @@ export const isOwner = (safeOwners: AddressEx[] | NamedAddress[] = [], walletAdd
...
-export const isModuleExecutionInfo = (
-  value: TransactionSummary['executionInfo'] | DetailedExecutionInfo,
-): value is ModuleExecutionInfo => value?.type === EXECUTION_INFO_TYPES.MODULE
-
-enum EXECUTION_INFO_TYPES {
-  MULTISIG = 'MULTISIG',
-  MODULE = 'MODULE',
-}
+export const isModuleDetailedExecutionInfo = (value?: DetailedExecutionInfo): value is ModuleExecutionDetails =>
+  value?.type === DetailedExecutionInfoType.MODULE
```
Comment (DiogoSoaress) [inline src/utils/transaction-guards.ts:63]: I've renamed these methods to improve understanding of its use

File: src/utils/transaction-guards.ts [hunk 2]
```diff
@@ -47,32 +54,15 @@ export const isOwner = (safeOwners: AddressEx[] | NamedAddress[] = [], walletAdd
   return safeOwners.some((owner) => sameAddress(owner.address, walletAddress))
 }
 
-export const isMultisigExecutionDetails = (value?: DetailedExecutionInfo): value is MultisigExecutionDetails => {
-  return value?.type === 'MULTISIG'
+// Narrows `TransactionDetails` -> when requesting a single transaction
```
Comment (iamacook) [inline src/utils/transaction-guards.ts:57]: The guards are for `TransactionDetails['detailedExecutionInfo']`.

File: src/utils/transaction-guards.ts [hunk 3]
```diff
@@ -129,6 +112,13 @@ export const isDateLabel = (value: TransactionListItem): value is DateLabel => {
   return value.type === TransactionListItemType.DATE_LABEL
 }
 
+// Narrows `Transaction`s -> when request a page of `TransactionListItem`
```
Comment (iamacook) [inline src/utils/transaction-guards.ts:115]: The guards are for `TransactionSummary['executionInfo']`.

**Review by iamacook — COMMENTED** [review]
I don't think the following comments are necessary but if we want to keep them, they aren't quite correct.

File: src/utils/transaction-guards.ts (hunk 2 above)
Comment (DiogoSoaress) [reply src/utils/transaction-guards.ts:57]: maybe the comments are unnecessary after all. I thought they would help clarifying the situations where the code deals with `ExecutionInfo` vs `DetailedExecutionInfo`

**Review by iamacook — APPROVED** [review]

**Pull request merged**